### ISSUE TITLE
Failure on FreeBSD: error: use of undeclared identifier 'WIFEXITED'

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#142]
**[Original report](https://bitbucket.org/agriggio/art/issues/142) by Yuri Vic (Bitbucket: [Yuri Vic](https://bitbucket.org/Yuri Vic), ).**

----------------------------------------

```

/usr/ports/graphics/art/work/agriggio-art-b2d77c8f0080/rtengine/subprocess.cc:285:15: error: use of undeclared identifier 'WIFEXITED'
        if (!(WIFEXITED(exit_status) && WEXITSTATUS(exit_status) == 0)) {
              ^
/usr/ports/graphics/art/work/agriggio-art-b2d77c8f0080/rtengine/subprocess.cc:285:41: error: use of undeclared identifier 'WEXITSTATUS'
        if (!(WIFEXITED(exit_status) && WEXITSTATUS(exit_status) == 0)) {
                                        ^
6 warnings and 2 errors generated.
```

```
sys/wait.h needs to be included:
```

```
$ grep -r WIFEXITED /usr/include/
/usr/include/sys/wait.h:#define	WIFEXITED(x)	(_WSTATUS(x) == 0)
```

‌

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Thanks for reporting. However, I cannot commit to supporting BSD, I just don’t have the manpower to do that. Patches are welcome though…

**Comment by agriggio** [comment]
**Original comment by Yuri Vic (Bitbucket: [Yuri Vic](https://bitbucket.org/Yuri Vic), ).**

----------------------------------------

Based on the manpage [https://man7.org/linux/man-pages/man2/waitid.2.html](https://man7.org/linux/man-pages/man2/waitid.2.html) Linux also requires inclusion of

```
       #include <sys/types.h>
       #include <sys/wait.h>
```

for WIFEXITED, WEXITSTATUS, etc.

‌

If it doesn’t complain during compilation - it passes it by luck because these headers happen to be included elsewhere.

‌

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

added missing #includes

Fixes #142
